File: branches/0.1.0-Fix/java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/PrintingSample.java
package org.pdfclown.samples.cli;

import java.awt.print.PrinterException;

import org.pdfclown.files.File;
import org.pdfclown.tools.Renderer;

/**
  This sample demonstrates <b>how to print a PDF document</b>.
  <p>Note: printing is currently in pre-alpha stage; therefore this sample is
  nothing but an initial stub (no assumption to work!).</p>

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.1.0
  @version 0.1.0
*/
public class PrintingSample
  extends Sample
{
  @Override
  public boolean run(
    )
  {
    String filePath = promptPdfFileChoice("Please select a PDF file");

    // 1. Open the PDF file!
    File file;
    try
    {file = new File(filePath);}
    catch(Exception e)
    {throw new RuntimeException(filePath + " file access error.",e);}

    // 2. Print the document!
    Renderer renderer = new Renderer();
    boolean silent = false;
    try
    {
      if(renderer.print(file.getDocument(), silent))
      {System.out.println("Print fulfilled.");}
      else
      {System.out.println("Print discarded.");}
    }
    catch(PrinterException e)
    {System.out.println("Print failed: " + e.getMessage());}

    return true;
  }
}

File: tags/0.1.0-Alpha/dotNET/pdfclown.samples.cli/src/org/pdfclown/samples/cli/PrintingSample.cs
using org.pdfclown.files;
using org.pdfclown.tools;

using System;
using System.Collections.Generic;
using System.Drawing;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to print a PDF document.<summary>
    <remarks>Note: printing is currently in pre-alpha stage; therefore this sample is
    nothing but an initial stub (no assumption to work!).</remarks>
  */
  public class PrintingSample
    : Sample
  {
    public override bool Run(
      )
    {
      string filePath = PromptPdfFileChoice("Please select a PDF file");

      // 1. Open the PDF file!
      File file = new File(filePath);

      // 2. Print the document!
      Renderer renderer = new Renderer();
      bool silent = false;
      if(renderer.Print(file.Document, silent))
      {Console.WriteLine("Print fulfilled.");}
      else
      {Console.WriteLine("Print discarded.");}

      return true;
    }
  }
}